packages/block-library/src/categories/index.php

```diff
@@ -99,20 +99,26 @@ function render_block_core_categories( $attributes, $content, $block ) {
  */
 function build_dropdown_script_block_core_categories( $dropdown_id ) {
 	ob_start();
+
+	$exports = array( $dropdown_id, home_url() );
 	?>
 	<script>
-	( function() {
-		var dropdown = document.getElementById( '<?php echo esc_js( $dropdown_id ); ?>' );
-		function onCatChange() {
-			if ( dropdown.options[ dropdown.selectedIndex ].value !== -1 ) {
-				location.href = "<?php echo esc_url( home_url() ); ?>/?" + dropdown.name + '=' + dropdown.options[ dropdown.selectedIndex ].value;
+	( ( [ dropdownId, homeUrl ] ) => {
+		document.getElementById( dropdownId ).addEventListener( 'change', ( event ) => {
+			const dropdown = /** @type {HTMLSelectElement} */ ( event.target );
+			if ( dropdown.value && dropdown.value !== '-1' ) {
+				const url = new URL( homeUrl );
+				url.searchParams.set( dropdown.name, dropdown.value );
+				location.href = url.href;
 			}
-		}
-		dropdown.onchange = onCatChange;
-	})();
+		} );
+	})( <?php echo wp_json_encode( $exports, JSON_HEX_TAG | JSON_UNESCAPED_SLASHES ); ?> );
```

Comment:
```suggestion
	} )( <?php echo wp_json_encode( $exports, JSON_HEX_TAG | JSON_UNESCAPED_SLASHES ); ?> );
```

Optional and only because you went to such efforts to adhere to the style guide in the code directly above 😄 

I guess the `JSON_PRETTY_PRINT` flag could be used to go the "whole hog", or encode each item in the array separately but I don't reckon it's that important!